src/code/z_message_PAL.c

```diff
@@ -1,66 +1,3285 @@
+                    msgCtx->msgBufDecoded[decodedBufPos] = digits[i] + '0';
+                    charTexIdx += FONT_CHAR_TEX_SIZE;
+                    decodedBufPos++;
+                }
+            }
+            decodedBufPos--;
+        } else if (temp_s2 == MESSAGE_TOKENS) {
+            // Convert the current number of collected gold skulltula tokens to digits and
+            //  add the digits to the decoded buffer in place of the control character.
+            // "Total number of gold stars"
+            osSyncPrintf("\n金スタ合計数 ＝ %d", gSaveContext.inventory.gsTokens);
+            digits[0] = digits[1] = 0;
+            digits[2] = gSaveContext.inventory.gsTokens;
+
+            while (digits[2] >= 100) {
+                digits[0]++;
+                digits[2] -= 100;
+            }
+            while (digits[2] >= 10) {
+                digits[1]++;
+                digits[2] -= 10;
+            }
+
+            loadChar = false;
+            for (i = 0; i < 3; i++) {
+                if (i == 2 || digits[i] != 0) {
+                    loadChar = true;
+                }
+                if (loadChar) {
+                    Font_LoadChar(font, digits[i] + '0' - ' ', charTexIdx);
+                    msgCtx->msgBufDecoded[decodedBufPos] = digits[i] + '0';
+                    charTexIdx += FONT_CHAR_TEX_SIZE;
+                    osSyncPrintf("%x(%x) ", digits[i] + '0' - ' ', digits[i]);
+                    decodedBufPos++;
+                }
+            }
+            decodedBufPos--;
+        } else if (temp_s2 == MESSAGE_FISH_INFO) {
+            // "Fishing hole fish size"
+            osSyncPrintf("\n釣り堀魚サイズ ＝ ");
+            digits[0] = 0;
+            digits[1] = gSaveContext.minigameScore;
+
+            while (digits[1] >= 10) {
+                digits[0]++;
+                digits[1] -= 10;
+            }
+
+            for (i = 0; i < 2; i++) {
+                if (i == 1 || digits[i] != 0) {
+                    Font_LoadChar(font, digits[i] + '0' - ' ', charTexIdx);
+                    msgCtx->msgBufDecoded[decodedBufPos] = digits[i] + '0';
+                    charTexIdx += FONT_CHAR_TEX_SIZE;
+                    osSyncPrintf("%x(%x) ", digits[i] + '0' - ' ', digits[i]);
+                    decodedBufPos++;
+                }
+            }
+            decodedBufPos--;
+        } else if (temp_s2 == MESSAGE_HIGHSCORE) {
+            phi_s0_3 = HIGH_SCORE((u8)font->msgBuf[++msgCtx->msgBufPos]);
+            // "Highscore"
+            osSyncPrintf("ランキング＝%d\n", font->msgBuf[msgCtx->msgBufPos]);
+            if ((font->msgBuf[msgCtx->msgBufPos] & 0xFF) == 2) {
+                if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
+                    phi_s0_3 &= 0x7F;
+                } else {
+                    osSyncPrintf("HI_SCORE( kanfont->mbuff.nes_mes_buf[message->rdp] & 0xff000000 ) = %x\n",
+                                 HIGH_SCORE(font->msgBufWide[msgCtx->msgBufPos] & 0xFF000000));
```

Comment:
I'm pretty sure we'll need the wide char buffer for japanese versions, so I think it's more appropriate to just do it right from the start, since even if I undid it now it'll just be added back later and the cast removed

Reply:
Alright, if it's supposed to be used for ntsc versions then it's probably fine to leave it